lib/content-services/src/lib/search/components/search-date-range-advanced-tabbed/search-date-range-advanced/search-date-range-advanced.component.ts

```diff
@@ -0,0 +1,163 @@
+import { Subject } from 'rxjs';
+import { endOfDay, parse, isValid, isBefore, isAfter } from 'date-fns';
+import { DateAdapter, MAT_DATE_FORMATS, MAT_DATE_LOCALE, MatDateFormats } from '@angular/material/core';
+import { DateFnsAdapter, MAT_DATE_FNS_FORMATS } from '@angular/material-date-fns-adapter';
+import { InLastDateType } from './in-last-date-type';
+import { DateRangeType } from './date-range-type';
+import { SearchDateRangeAdvanced } from './search-date-range-advanced';
+import { FormBuilder, UntypedFormControl, Validators } from '@angular/forms';
+import { takeUntil } from 'rxjs/operators';
+import { UserPreferencesService, UserPreferenceValues, DateFnsUtils } from "@alfresco/adf-core";
+
+const DEFAULT_DATE_DISPLAY_FORMAT = 'dd-MMM-yy';
+
+@Component({
+    selector: 'adf-search-date-range-advanced',
+    templateUrl: './search-date-range-advanced.component.html',
+    styleUrls: ['./search-date-range-advanced.component.scss'],
+    providers: [
+        { provide: DateAdapter, useClass: DateFnsAdapter, deps: [ MAT_DATE_LOCALE ] },
+        { provide: MAT_DATE_FORMATS, useValue: MAT_DATE_FNS_FORMATS }
+    ],
+    encapsulation: ViewEncapsulation.None,
+    host: {class: 'adf-search-date-range-advanced'}
+})
+export class SearchDateRangeAdvancedComponent implements OnInit, OnDestroy {
+    @Input()
+    dateFormat = DEFAULT_DATE_DISPLAY_FORMAT;
+    @Input()
+    maxDate: string;
+    @Input()
+    field: string;
+    @Input()
+    set initialValue(value: SearchDateRangeAdvanced) {
+        if (value) {
+            this.form.patchValue(value);
+        }
+    }
+
+    @Output()
+    changed = new EventEmitter<Partial<SearchDateRangeAdvanced>>();
+    @Output()
+    valid = new EventEmitter<boolean>();
+
+    form = this.formBuilder.group<SearchDateRangeAdvanced>({
+        dateRangeType: DateRangeType.ANY,
+        inLastValueType: InLastDateType.DAYS,
+        inLastValue: undefined,
+        betweenStartDate: undefined,
+        betweenEndDate: undefined
+    });
+    betweenStartDateFormControl = this.form.controls.betweenStartDate;
+    betweenEndDateFormControl = this.form.controls.betweenEndDate;
+    convertedMaxDate: Date;
+    private destroy$ = new Subject<void>();
+
+    readonly DateRangeType = DateRangeType;
+    readonly InLastDateType = InLastDateType;
+
+    constructor(private formBuilder: FormBuilder,
+                private userPreferencesService: UserPreferencesService,
+                private dateAdapter: DateAdapter<DateFnsAdapter>,
+                @Inject(MAT_DATE_FORMATS) private dateFormatConfig:MatDateFormats) {}
+
+    readonly endDateValidator = (formControl: UntypedFormControl): ({ [key: string]: boolean } | null) => {
+        if (isBefore(formControl.value, this.betweenStartDateFormControl.value) || isAfter(formControl.value, this.convertedMaxDate)) {
+            return {
+                invalidDate: true
+            };
+        }
+        return {};
+    }
```

Comment:
missing semicolon

Reply:
These has been resolved in [this](https://github.com/Alfresco/alfresco-ng2-components/pull/8714/commits/bdaa0e52fd3a31ba7c3802c6cdc6a27a733d6057) commit